packages/web-components/src/components/ic-badge/ic-badge.tsx

```diff
@@ -193,19 +195,51 @@ export class Badge {
   };
 
   private setPositionInTopNavigation = () => {
-    const parentTopNavEl = this.el.parentElement?.parentElement;
-    parentTopNavEl?.classList.contains("mobile-mode")
-      ? (this.position = "inline")
-      : (this.position = "near");
+    this.position = this.isInMobileMode() ? "inline" : this.initialPosition;
+  };
+
+  private setPositionInSideNavigation = () => {
+    this.position = "near";
   };
 
   private isInTopNav = (): boolean => {
     const parentEl = this.el.parentElement;
     if (parentEl) {
-      const grandparentEl = parentEl?.parentElement;
+      const grandparentEl = parentEl.parentElement;
```

Comment:
you can invert this if statement. `if(!parentEl) return false;` and then have your logic outside